impl/src/bitfield.rs

```diff
@@ -90,7 +82,8 @@ impl BitfieldStruct {
 
             impl #ident
             {
-                pub fn new() -> Self {
+                /// Returns an instance with zero initialized data
+                fn new() -> Self {
```

Comment:
Why make it always private? From what I recall we have tests that directly use the `new` constructor which would probably fail if they can no longer call it. I am sure we should maybe implement `Default` here instead of defining a `new` constructor.

Reply:
`Inherited` still means that the crate defining the type will be able access it and i.e. add a meaningful constructor on top. Using `Default` might be smarter, on the other hand that can be derived by the user via `#[derive(Default)]` if useful. The point is that if this kind of zero-initializing new is public, you can't enforce invariants (even less so if `to_bytes` is public as well) without wrapping the bitfield type. Keep in mind that the module using `#[bitfield]` always has access. Maybe `pub(crate)` would be a good middle ground, but that is then just arbitrary.

Reply:
Maybe it should inherit the visibility of the `struct` itself. If the `struct` is `pub` its constructor is too and vice versa.